apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx

```diff
@@ -286,37 +288,26 @@ export default function SectionsSetUpContainer({
           )}
         </div>
       </div>
-
       <div
         className={classnames(
-          moduleStyles.buttonsContainer,
+          moduleStyles.splitButtonsContainer,
           moduleStyles.containerWithMarginTop
         )}
       >
-        {/*
-         TODO: for the first iteration of this feature we will only have
-        participants create one section at a time.  For edit section redirects,
-        adding another section is not needed.  This can be uncommented when the
-        functionality of adding another class section is ready.
-
-        {isNewSection && (
-        <Button
-          useDefaultLineHeight
-          icon="plus"
-          text={i18n.addAnotherClassSection()}
-          color={Button.ButtonColor.neutralDark}
-          onClick={e => {
-            e.preventDefault();
-            console.log('Add Another Class Section clicked');
-          }}
-        />
-        */}
-        {/*
-        TODO: currently this button just changes text if it is a "new" or "editied"
-        screen, depending on how we want the functionality of this button to work,
-        this might mean creating a different button for the "edit" page
-        */}
+        {isNewSection && ( // Only show 'save and add another' button when creating a new section
+          <Button
+            className={moduleStyles.buttonLeft}
+            icon="plus"
+            text={i18n.addAnotherClassSection()}
+            color={Button.ButtonColor.neutralDark}
+            onClick={e => {
+              e.preventDefault();
```

Comment:
I've tried fixing it locally and it looks like it might be deeper than this. First of all, do we really need to store array of sections? From what I see we only using one section constantly.

Anyways, I think it would be awesome to wrap callbacks into useCallback hook. Also I guess it might be great to refactor the container logic so it would be splitter to separate files and the change the data structure so there'll be no unneeded re-renders.

Reply:
I agree that this would be a good refactor- we have some time set aside to revisit and do cleanup. I'll add this note! Fwiw- we initially designed this to handle multiple sections at a time, and hope to come back to that work soon.

Reply:
I created the ticket here: https://codedotorg.atlassian.net/browse/TEACH-532

Reply:
oh, gotcha! Awesome, thank you!